topeft/modules/datacard_tools.py

```diff
@@ -306,6 +307,29 @@ def __init__(self,pkl_path,**kwargs):
         with open(topeft_path("params/wc_ranges.json"), "r") as wc_ranges_json:
             self.wc_ranges = json.load(wc_ranges_json)
 
+        with open(topeft_path("params/SMEFTsim-topU3l_dim6top.yml")) as f:
+            self.rotate = yaml.load(f,Loader=yaml.CLoader)
+
+        for k,v in self.rotate.items():
+            if isinstance(v, str):
+                if "CW" in v or "SW" in v:
+                    v = eval(v.replace("{CW}", str(self.rotate["CW"])).replace("{SW}", str(self.rotate["SW"])))
+                    self.rotate[k] = v
+            elif isinstance(v, list):
+                for i,vv in enumerate(v):
+                    if "CW" in vv or "SW" in vv:
+                        v[i] = vv.replace("{CW}", str(self.rotate["CW"])).replace("{SW}", str(self.rotate["SW"]))
+                wcs = {}
+                for wc in v[1]:
+                    lo,hi = self.wc_ranges[wc]
+                    wcs[wc] = f"{wc}[0,{lo},{hi}]"
+                v = v[0]
+                self.rotate[k] = v
```

Comment:
I think you accidentally removed the `v = v.format(**wcs)` line, which is still needed. So it should be:
```python
wcs = {}
for wc in v[1]:
    lo,hi = self.wc_ranges[wc]
    wcs[wc] = f"{wc}[0,{lo},{hi}]"
v = v[0]
v = v.format(**wcs)
self.rotate[k] = v
```

Reply:
Woops, should be fixed now